### PR TITLE
pkgdown setup

### PR DESCRIPTION
**Pull request by radbasa**
Extends from PR #66 

Issue #43

## Description
* set up `pkgdown`. ready for vignettes.

## Definition of Done
- [ ] The change is thoroughly documented.
- [ ] The CI passes (`R CMD check`, linter, unit tests, spelling).
- [ ] Any generated files have been updated (e.g. `.Rd` files with `roxygen2::roxygenise()`)

### CODE REVIEW

File: pkgdown/_pkgdown.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
+title: box.linters
+template:
+  bootstrap: 5
+  bootswatch: pulse
+  bslib:
+    pkgdown-nav-height: 100px
+  includes:
+    in_header: |
+      <script async defer src="https://buttons.github.io/buttons.js"></script>
```
Comment (jakubnowicki) [inline pkgdown/_pkgdown.yml:9]:
```suggestion
    in_header: |
    <!-- Google tag (gtag.js) -->
      <script async src="https://www.googletagmanager.com/gtag/js?id=G-8GE9YEGHF6"></script>
      <script>
        window.dataLayer = window.dataLayer || [];
        function gtag(){dataLayer.push(arguments);}
        gtag('js', new Date());
        gtag('config', 'G-DVX9BJSBX4');
      </script>
      <!-- Google tag (gtag.js) -->
      <script async src="https://www.googletagmanager.com/gtag/js?id=G-FQQZL5V93G"></script>
      <script>
        window.dataLayer = window.dataLayer || [];
        function gtag(){dataLayer.push(arguments);}
        gtag('js', new Date());
        gtag('config', 'G-FQQZL5V93G');
      </script>
      <script async defer src="https://buttons.github.io/buttons.js"></script>
```

File: pkgdown/_pkgdown.yml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    before_navbar: |
+      <a href="https://rhinoverse.dev" class="home"><i class="fa fa-lg fa-chevron-left"></i></a>
+url: https://appsilon.github.io/box.linters/
+
+navbar:
+  structure:
```
Comment (jakubnowicki) [inline pkgdown/_pkgdown.yml:15]:
```suggestion
navbar:
  type: light
  structure:
```

File: pkgdown/extra.css [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+#navbar > ul.navbar-nav > li.nav-item a:hover {
+  background-color: var(--primary-color) !important;
+}
+
+#navbar > ul.navbar-nav > li.nav-item > a {
+  color: rgba(255, 255, 255, 0.8);
+}
+
+.navbar-dark .navbar-nav .active > .nav-link {
```
Comment (jakubnowicki) [inline pkgdown/extra.css:17]:
```suggestion
.navbar-light .navbar-nav .active > .nav-link {
```

File: pkgdown/extra.css [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+.navbar-brand {
+  color: #fff !important;
+}
+
+.navbar-dark input[type="search"] {
+  background-color: #fff !important;
+  color: #444 !important;
+}
+
```
Comment (jakubnowicki) [inline pkgdown/extra.css:30]:
```suggestion
```
Input search style is fixed by `type: light` in `_pkgdown.yml`

**Review by jakubnowicki — CHANGES_REQUESTED** [review]
Please check my comments. Also, please add [the CI configuration for `pkgdown`](https://github.com/Appsilon/reactable.extras/blob/main/.github/workflows/pkgdown.yml)

**Comment by radbasa** [comment]
> Please check my comments. Also, please add [the CI configuration for `pkgdown`](https://github.com/Appsilon/reactable.extras/blob/main/.github/workflows/pkgdown.yml)

Used `pkgdown.yaml` CI from `usethis::use_github_actions("pkgdown")`

**Review by jakubnowicki — APPROVED** [review]
LGTM

**Pull request merged**